libimage/filters.go

```diff
@@ -19,13 +19,15 @@ import (
 
 // filterFunc is a prototype for a positive image filter.  Returning `true`
 // indicates that the image matches the criteria.
-type filterFunc func(*Image) (bool, error)
+type filterFunc func(*Image, *layerTree) (bool, error)
+
+type filtersMap map[string][]filterFunc
```

Comment:
(Non-blocking: This might be `compiledFilters`. “Map/Array/…” in a type name is not all that valuable.

Alternatively, `compiledFilters` could be a
```go
struct {
   filters map…
   needsLayerTree bool
}
```
but it’s not really necessary to build that much abstraction for a single caller — and it might be inconvenient for the way we use `needsLayerTree` in `ListImages`.)